chore(stdlib)!: Replace Number.nan & Number.infinity constants with keywords

phated opened the pull request:
This removes the `Number.nan` and `Number.infinity` constants and uses `Infinity` and `NaN` keywords. Ideally, this sort of changes will happen with the feature so we aren't adding 2 different breaking changes for it in the changelog.

ospencer reviewed (commented):
I didn't think we needed to get rid of the constants in the library, but I guess they don't serve much of a purpose if we have the literals 🤔

phated commented:
@ospencer I was rewriting the tests and found that there was a ridiculous matrix of testing variations on using the keywords and constants. It doesn't make sense to have more than one way to access `Infinity` and `NaN` so I removed them and updated everything instead.

ospencer reviewed (approved):
Looks good, just needs a Graindoc update.
